app/api/domains/api.py

```diff
@@ -107,6 +109,40 @@ def format_map_basic(m: Beatmap) -> dict[str, object]:
         "diff": m.diff,
     }
 
+@router.get("/search")
+async def api_search(
+    search: Optional[str] = Query(None, alias='src'),
+    db_conn: databases.core.Connection = Depends(acquire_db_conn),
+):
+    """Search for users on the server by name."""
+    
+    # we're using text() here because we are going to
+    # assign placeholders (bindparams()) to these queries, 
+    # and this will let us bind those safely and efficiently.
+    results_query = text("SELECT COUNT(id) FROM users WHERE name AND priv >= 3 LIKE :search_clause") # we will only need to lookup users who have Privileges.VERIFIED
+    rows_query = text("SELECT id, name FROM users WHERE name LIKE :search_clause AND priv >= 3 ORDER BY id ASC") # to save on host resources and prevent useless requests
+
+    # using parameterised queries helps prevent against sql injections
+    # and will also allow the database engine to cache our queries,
+    # making these lookups faster in the future
+    search_clause = f"%{search}%" if search else None
+    results_query = results_query.bindparams(search_clause=search_clause)
+    rows_query = rows_query.bindparams(search_clause=search_clause)
+
+    # execute the queries using the database connection
+    results = await db_conn.fetch_val(results_query)
+    rows = await db_conn.fetch_all(rows_query)
+
+    # return the response
+    response = ORJSONResponse(
```

Comment:
why are you assigning the response instead of just retuurning it?

Reply:
this was already changed in the last commit

Reply:
Check out the code I sent in general on the Discord, I'd suggest replacing this whole PR with that code

Reply:
my code* :trollface:

Reply:
https://discord.com/channels/748687781605408908/748687781605408911/1056576143668555907

Reply:
> my code* :trollface:

Didn't say it's mine

Reply:
no